### PR TITLE
routing-api simulation status change from string to enum

### PR DESCRIPTION
**Pull request by jsy1218**
trading api team requested to change the simulation status type from string to enum. 

we will still return the `simulationStatus` response field as string, as tested in my local https://app.warp.dev/block/Z8Iu8aUYWn7F3hxXSIQh91.

All e2e tests can pass https://app.warp.dev/block/pHRc3WeTfygM4wgYHEmHSP, indicating no change of the `simulationStatus` response field string type.

### CODE REVIEW

**Comment by jsy1218** [comment]
* **#757** <a href="https://app.graphite.dev/github/pr/Uniswap/routing-api/757?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jsy1218 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/handlers/quote/util/simulation.ts [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
+      return RoutingApiSimulationStatus.INSUFFICIENT_BALANCE
     case SimulationStatus.NotSupported:
-      return 'NOT_SUPPORTED'
+      return RoutingApiSimulationStatus.NOT_SUPPORTED
     case SimulationStatus.NotApproved:
-      return 'NOT_APPROVED'
+      return RoutingApiSimulationStatus.NOT_APPROVED
     default:
       log.error(`Unknown simulation status ${simulationStatus}`)
-      return ''
```
Comment (jsy1218) [inline lib/handlers/quote/util/simulation.ts:20]: there's a behaviorial change from returning empty string '' to UNKNOWN enum type in the response. trading-api team confirmed this is ok. Will check in URA, as well as with web team.

Comment (jsy1218) [reply lib/handlers/quote/util/simulation.ts:20]: web team confirmed this is ok.

Comment (jsy1218) [reply lib/handlers/quote/util/simulation.ts:20]: URA uses `simulationStatus` as pass-through https://github.com/search?q=repo%3AUniswap%2Funified-routing-api%20simulationStatus&type=code

Comment (jsy1218) [reply lib/handlers/quote/util/simulation.ts:20]: all ok

**Review by cgkol — APPROVED** [review]
LG! assuming both web and trading api can handle we are good to go

**Comment by jsy1218** [comment]
### Merge activity

* **Jun 20, 2:15 PM EDT**: @jsy1218 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/Uniswap/routing-api/757).

**Pull request merged**